docs/contributor_guide/testing.md

```diff
@@ -0,0 +1,29 @@
+# Testing
+
+## Python unit tests
+
+Requires `pytest` and `dirty-equals`:
+
+```bash
+pip install pytest dirty-equals
+```
+
+Run the tests:
+
+```bash
+pytest --color=yes -v python
+```
+
+## UI tests
+
+We don't recommend running UI tests locally because they are browser-based tests and
+differences between systems can cause test breakage.
+Integration tests will run automatically in CI.
+
+However, if you are developing a new UI test or editing an existing UI test, you may
+want to run them locally:
+
+```bash
+npx playwright install chromium  # Install testing browser
+npx playwright test --retries=2  # Run tests
```

Comment:
```suggestion
jlpm playwright install chromium  # Install testing browser
jlpm run test  # Run tests
```

Lets recommend using `jlpm` :)

Reply:
Ah sorry I forgot to mention, one would have to change directory to `ui-tests` too

Reply:
Makes sense, thank you! Do you think we should also update our GitHub actions to run using `jlpm`? That's where I got the idea to run with `npx`. Perhaps it's that way in GHA for caching reasons.

ac079b7

Reply:
yeah let's not change GHA in this PR, although i'm happy to take this discussion in a separate issue.